### PR TITLE
support build with go 1.18

### PR DESCRIPTION
**Pull request by freeznet**
### Motivation

Build pulsarctl with go 1.18 failed with following error:

```
# golang.org/x/sys/unix
/Users/rui/go/pkg/mod/golang.org/x/sys@v0.0.0-20201119102817-f84b799fce68/unix/syscall_darwin.1_13.go:29:3: //go:linkname must refer to declared function or variable
/Users/rui/go/pkg/mod/golang.org/x/sys@v0.0.0-20201119102817-f84b799fce68/unix/zsyscall_darwin_amd64.1_13.go:27:3: //go:linkname must refer to declared function or variable
/Users/rui/go/pkg/mod/golang.org/x/sys@v0.0.0-20201119102817-f84b799fce68/unix/zsyscall_darwin_amd64.1_13.go:40:3: //go:linkname must refer to declared function or variable
/Users/rui/go/pkg/mod/golang.org/x/sys@v0.0.0-20201119102817-f84b799fce68/unix/zsyscall_darwin_amd64.go:28:3: //go:linkname must refer to declared function or variable
/Users/rui/go/pkg/mod/golang.org/x/sys@v0.0.0-20201119102817-f84b799fce68/unix/zsyscall_darwin_amd64.go:43:3: //go:linkname must refer to declared function or variable
/Users/rui/go/pkg/mod/golang.org/x/sys@v0.0.0-20201119102817-f84b799fce68/unix/zsyscall_darwin_amd64.go:59:3: //go:linkname must refer to declared function or variable
/Users/rui/go/pkg/mod/golang.org/x/sys@v0.0.0-20201119102817-f84b799fce68/unix/zsyscall_darwin_amd64.go:75:3: //go:linkname must refer to declared function or variable
/Users/rui/go/pkg/mod/golang.org/x/sys@v0.0.0-20201119102817-f84b799fce68/unix/zsyscall_darwin_amd64.go:90:3: //go:linkname must refer to declared function or variable
/Users/rui/go/pkg/mod/golang.org/x/sys@v0.0.0-20201119102817-f84b799fce68/unix/zsyscall_darwin_amd64.go:105:3: //go:linkname must refer to declared function or variable
/Users/rui/go/pkg/mod/golang.org/x/sys@v0.0.0-20201119102817-f84b799fce68/unix/zsyscall_darwin_amd64.go:121:3: //go:linkname must refer to declared function or variable
/Users/rui/go/pkg/mod/golang.org/x/sys@v0.0.0-20201119102817-f84b799fce68/unix/zsyscall_darwin_amd64.go:121:3: too many errors
make: *** [pulsarctl] Error 2
```

According to https://github.com/golang/go/issues/49219, need to update `golang.org/x/sys` to the latest to fix this issue.

### Modifications

*Describe the modifications you've done.*

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

Check the box below.

Need to update docs? 

- [ ] `doc-required` 
  
  (If you need help on updating docs, create a doc issue)
  
- [x] `no-need-doc` 
  
  (Please explain why)
  
- [ ] `doc` 
  
  (If this PR contains doc changes)

### CODE REVIEW

File: .github/workflows/ci-bookie-checks.yml [hunk 1]
```diff
@@ -14,7 +14,7 @@ jobs:
     - name: Set up Go 1.13
```
Comment (nodece) [inline .github/workflows/ci-bookie-checks.yml:14]:
```suggestion
    - name: Set up Go 1.18
```

File: .github/workflows/ci-functions-checks.yml [hunk 2]
```diff
@@ -14,7 +14,7 @@ jobs:
     - name: Set up Go 1.13
```
Comment (nodece) [inline .github/workflows/ci-functions-checks.yml:14]:
```suggestion
    - name: Set up Go 1.18
```

File: .github/workflows/ci-functions-checks.yml [hunk 3]
```diff
@@ -26,7 +26,7 @@ jobs:
       - name: Set up Go 1.13
```
Comment (nodece) [inline .github/workflows/ci-functions-checks.yml:26]:
```suggestion
      - name: Set up Go 1.18
```

File: .github/workflows/ci-functions-checks.yml [hunk 4]
```diff
@@ -38,7 +38,7 @@ jobs:
...
```
Comment (nodece) [inline .github/workflows/ci-functions-checks.yml:38]:
```suggestion
      - name: Set up Go 1.18
```

File: .github/workflows/ci-packages-checks.yml [hunk 5]
```diff
@@ -14,7 +14,7 @@ jobs:
       - name: Set up Go 1.13
```
Comment (nodece) [inline .github/workflows/ci-packages-checks.yml:14]:
```suggestion
      - name: Set up Go 1.18
```

File: .github/workflows/ci-style-checks.yml [hunk 6]
```diff
@@ -14,7 +14,7 @@ jobs:
       - name: Set up Go 1.13
```
Comment (nodece) [inline .github/workflows/ci-style-checks.yml:14]:
```suggestion
      - name: Set up Go 1.18
```

File: .github/workflows/ci-release-checks.yml [hunk 7]
```diff
@@ -16,7 +16,7 @@ jobs:
       - uses: actions/checkout@v2
       - uses: actions/setup-go@v1
         with:
-          go-version: 1.13
+          go-version: 1.18
```
Comment (nodece) [inline .github/workflows/ci-release-checks.yml:19]: Could you add a build matrix? You need to test 1.13, 1.14, 1.15, 1.16, 1.17, 1.18.

File: pkg/pulsar/utils/namespace_name.go [hunk 8]
```diff
@@ -84,10 +84,7 @@ func validateNamespaceName(tenant, namespace string) error {
 const PATTEN = "^[-=:.\\w]*$"
 
 func CheckName(name string) bool {
-	patten, err := regexp.Compile(PATTEN)
-	if err != nil {
-		return false
-	}
+	patten := regexp.MustCompile(PATTEN)
```
Comment (nodece) [inline pkg/pulsar/utils/namespace_name.go:87]: Move this to line 85.

File: pkg/ctl/functions/upload_test.go [hunk 9]
```diff
@@ -128,6 +125,8 @@ func TestUploadAndDownloadCommands(t *testing.T) {
 		log.Fatal(err)
 		t.Fail()
 	}
+	defer os.RemoveAll(f.Name())
```
Comment (nodece) [inline pkg/ctl/functions/upload_test.go:128]: Old behavior looks better.

Comment (freeznet) [reply pkg/ctl/functions/upload_test.go]: use `log.Panic` instead of `log.Fatal`, since the `log.Fatal` will skip `defer` calls.

Comment (nodece) [reply pkg/ctl/functions/upload_test.go]: Good catch!

File: site/gen-pulsarctldocs/generators/gen.go [hunk 10]
```diff
@@ -259,41 +260,37 @@ func WriteCategoryFile(c Category) {
...
+		o.Usage = strings.ReplaceAll(o.Usage, "|", "&#124;")
 	}
 	for _, sc := range params.SubCommands {
 		for _, o := range sc.Options {
-			o.Usage = strings.Replace(o.Usage, "|", "&#124;", -1)
+			o.Usage = strings.ReplaceAll(o.Usage, "|", "&#124;")
 		}
 	}
 	f, err := os.Create(*GenPulsarctlDir + "/includes/_generated_" + strings.ToLower(params.MainCommand.Name) + ".md")
+	defer f.Close()
```
Comment (nodece) [inline site/gen-pulsarctldocs/generators/gen.go:286]: The `f` may be `nil`.

**Review by nodece — APPROVED** [review]
LGTM

**Pull request merged**